tests/Shared/ScopedSanityCheck.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Avalonia.UnitTests;
+using Xunit;
+
+namespace Avalonia.UnitTests.Helpers;
+
+static class ScopedSanityCheck
+{
+    [ModuleInitializer]
```

Comment:
Not really important, but could be replaced with an assembly attribute inheriting `BeforeAfterTestAttribute` to have nicer results (the non-scoped tests would fail and not all the others).

Reply:
Non-scoped tests are specifically NOT desired to be in the same assembly as scoped ones since they can pollute the global state too via implicit side effects.

This will be important later since `new AvaloniaObject()` would initialize a dispatcher bound to the current thread once we have multi-dispatcher support later.

Reply:
Note that ScopedTestBase also has a Collection attribute that disables test parallelism. Which is also very much desired in assemblies with tests for the code that utilize global state.

Reply:
Ah, can do that. The current approach was rather useful to get the full list of types to update.